Xamarin.Forms.Platform.iOS/CollectionView/ObservableItemsSource.cs

```diff
@@ -10,17 +10,21 @@ internal class ObservableItemsSource : IItemsViewSource
 	{
 		readonly UICollectionView _collectionView;
 		readonly IList _itemsSource;
+		bool _isEmpty;
+
 
 		public ObservableItemsSource(IEnumerable itemSource, UICollectionView collectionView)
 		{
 			_collectionView = collectionView;
 			_itemsSource = (IList)itemSource;
 
 			((INotifyCollectionChanged)itemSource).CollectionChanged += CollectionChanged;
+			_isEmpty = !_itemsSource.GetEnumerator().MoveNext();
 		}
 
 		void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
 		{
+			ShouldDetermineCellSize();
```

Comment:
We are adding our first cell, the collection was empty and now we will need to DetermineCellSize